src/game/updateGame.js

```diff
@@ -0,0 +1,21 @@
+export const updateGame = (
+  showAlert,
+  closeAlert,
+  cellValue,
+  boardData,
+  currentPlayer,
+  changeTurn,
+  updateBoard
+) => {
+  let updatedBoard = boardData;
+
+  if (cellValue !== "X" && cellValue !== "O") {
+    closeAlert();
+    updatedBoard = updateBoard(boardData, cellValue, currentPlayer);
+    changeTurn();
+  } else {
+    showAlert();
+  }
+
+  return updatedBoard;
```

Comment:
A couple refactor suggestions for later (I'm writing them down as they come to me to let you see my thought process of how I might refactor this) :

First, I would try to avoid using `let` if possible. But if you're going to use it, maybe set `updatedBoard = boardData` inside the `else` to better communicate that you want the original board? i actually missed it when you initialized it on line 10. e.g. 

```
let updatedBoard;

  if (cellValue !== "X" && cellValue !== "O") {
    updatedBoard = updateBoard(boardData, cellValue, currentPlayer);
    closeAlert();
    changeTurn();
  } else {
    updatedBoard = boardData;
    showAlert();
  }

  return updatedBoard;
```

Reply:
other potential rewrite: 

```
  if (cellValue === "X" && cellValue === "O") {
    showAlert()
    return boardData
  } else {
    closeAlert();
    changeTurn();
    return updateBoard(boardData, cellValue, currentPlayer);
  }
```
tbh, not quite a fan of this, but it gets rid of the `let` and communicates that the return value for each condition is different.

Reply:
Last suggestion, what if you move the responsibility to use the original board vs updating the board directly into updateBoard? e.g. 

updateGame.js:
```
 if (cellValue === "X" && cellValue === "O") {
    showAlert()
  } else {
    closeAlert();
    changeTurn();
  }
  
  return updateBoard(boardData, cellValue, currentPlayer);
```

updateBoard.js:
```
export function updateBoard(board, value, turn) {

if (value !== "X" && value !== "O") {
  return board
}

return board.map((row) =>
    row.map((cell) => (cell === value ? turn : cell))
  );
}
```

if you do this, then I think you can just call`updateBoard()` directly in Board.jsx > handleClick() instead of passing it down into `updateGame()`. It just feels like `updateGame` is doing a lot.

Reply:
just some ideas. curious if @ranizilpelwar  or @fifikim have thoughts on this.

Reply:
I know what you mean. It is a lot of parameters. After going over many different refactor options on this, I am still trying to figure out how to minimize the params being passed into updateGame. Most of this is because I'm keeping logic out of the Game component. If I weren't doing this, there would be no need to call an updateGame function or pass so many parameters into it. I am wondering if you have any other suggestions after this most recent refactor.